### DISCUSSION TITLE
Listen to migration event

### DISCUSSION THREAD

**Discussion by budirec** [adonisjs/core#3858] [Help]
Anyone know how to listen to migration event?

I would like to send email when a migration was ran and include the migration script that was ran.

I tried to add event listener for migration:start, but it catch nothing

```
Event.on('migration:start', function ({ file }) {
  Logger.debug(file)
})
```

**Comment by thetutlage** [comment]
Currently no public events are emitted when migrations are executed. Maybe, you can do it by creating a custom command. Something as follows.

```ts
node ace migration:run && node ace migration:notify
```

Next will be to create the `migration:notify` command. To keep things easier, you can put it all inside an npm script.

```json
{
  "scripts": {
    "migrate": "node ace migration:run && node ace migration:notify"
  }
}
```

And now run it as `npm run migrate`

**Reply by budirec** [reply]
If we do this, we can't be sure which migration was ran successfully, right?

**Reply by thetutlage** [reply]
Yup, there is no easy way to find which migrations were successful. I think we can emit global events for introspection. Can you please create an issue in the Lucid repo for the same?
